lib/lite_cable/iodine.rb

```diff
@@ -0,0 +1,179 @@
+# coding: utf-8
+# frozen_string_literal: true
+
+# TODO:
+# 1. broadcast работает только изнутри сервера Iodine(т.е. если хочется его из экшенов
+# синатры, а не из канала, то надо синатру и каналы запускать сервером iodine).
+# Заработает снаружи, если сделать что-то из этого:
+# a) сделать http путь типо /broadcast изнутри iodine rack.
+# b) запустить и синатру и каналы через сервер iodine(т.е. пуму убрать)
+# c) юзать Redis как pubsub engine И сделать доп. фикс -
+# или делать руками PUBLISH в redis,
+# или если Iodine научится запускаться без запуска сервера, как клиент к pubsub'у.
+#
+# 2. какие-то траблы с памятью при использовании одновременно с синатрой
+# https://github.com/boazsegev/iodine/issues/27
+#
+# 3. puts не работает внутри rack и сообщений с вебсокетов.
+
+module LiteCable # :nodoc:
+  # Iodine extensions
+  module Iodine
+    require "iodine"
+    # LiteCable::Server needs for subscribers_map -
+    # because Iodine don't have unsubscribe_from_all feature
+    # and can broadcast only for whole stream, without specific channel_id in message
+    require "lite_cable/server"
+
+    BROADCASTING_STREAM = :litecable_broadcasting
+
+    module Server # :nodoc:
+      class << self
+        # Iodine runs multiple processes, each of which stores own subscriptions_map -
+        # which sockets are subscribed for which channels and streams.
+        # Iodine.publish sends an message to all server processes, and here they make
+        # broadcast to their own sockets.
+        def setup_broadcast
+          ::Iodine.subscribe(channel: LiteCable::Iodine::BROADCASTING_STREAM) do |_ch, json_msg|
+            msg = JSON.parse(json_msg)
+            LiteCable::Server.broadcast(msg["stream"], msg["payload"])
+          end
+        end
+      end
+    end
+
+    module Broadcasting # :nodoc:
+      def broadcast(stream, message, coder: nil)
+        # TODO: судя по всему в Anycable тоже ожидается только один кодер - json. Иначе
```

Comment:
Yep, the existing server (nor clients) don't support other coders; but nevertheless it's better to consider them from the beginning in order to avoid upgrades in the future.

Reply:
Okay, got it. I added serialization if the coder is not same as default